CHANGELOG.md

```diff
@@ -1,3 +1,32 @@
+## 1.10.1-SNAPSHOT (YYYY-MM-DD)
+
+### Breaking Changes
+* None.
+
+### Enhancements
+* [Sync] Optimized the opening of Flexible Sync Realms when `waitForInitialRemoteData` is used. (Issue [#1438](https://github.com/realm/realm-kotlin/issues/1438))
```

Comment:
Do we need to put in a warning about that it is now required to be online when opening a FLX Realm the first time. Or maybe there would be some other condition/side effect of subscriptions, etc. that already essentially caused same constraint?

Reply:
The new behavior still depends on `waitForInitialRemoteData`, since we are also writing the subscriptions and making sure they were valid before continuing, the end user will still see the same timeout if disconnected, it will just happen earlier in our internal code now.

Reply:
Ahh, yes. Seems to be in `initializeRealmData`.

Reply:
Yes, I agree the flow is a bit confusing. It is an unfortunate side-effect of combining Sync And Local Realms in the same bootstrap flow + taking into account the various bootstrap options for both.